commitments/eigenda.go

```diff
@@ -17,8 +19,9 @@ type CertCommitment interface {
 }
 
 type EigenDACommitment struct {
-	prefix EigenDACommitmentType
-	b      []byte
+	prefix   EigenDACommitmentType
+	b        []byte
+	encoding *EncodingType // Optional encoding type, used only for V2+
```

Comment:
move above b to align with serialization order.
Also the EigenDACommitmentType is defined as a byte, whereas EncodingType is a string. Feel like we should use the same structure (which is also why they should both be defined in the same file imo, which makes it easier to see this structure.

Reply:
also don't think encoding should be optional? we always know what the encoding is (even if only implicitly in the case for v0 and v1 certs.

Reply:
agree

Reply:
04dba8c7e4368cfd65fdfc4b9bb95f480a8ec008